UI Restructuring

nseidm1 opened the pull request:
ActionBar Sherlock will be needed to build the final apk. The backend stuff is identical in functionality with major changed to the design and UI patterns. Feel welcome to reach out if you have any questions, I'll be happy to explain anything. Also feel welcome to cherry pick the changes. Ultimately I'd like to fold everything upstream so the fork will not be needed at all.

If you want to see the changes in action I published the fork to the Play Store for the time being. MetaWatch Mgr - Noah Edition will be the name when the Play Store gets it out there.

Commit:
Updated to a fragment based design using Action Bar Sherlock, a standard
dark holo theme, and a simplified main UI. Backend services are
identical, including all widget management etc...

nseidm1 committed:
Refresh the widget previews, and list every 3 seconds.

nseidm1 committed:
Realtime updates of the statistics dialog, fixed the connection
notification's pending intent to open the new main activity.

nseidm1 committed:
Version, icon, and proguard config.

Commit:
Reduced the size of the screen preview. Changed the connected
disconnected notification images. Increase the poll rate of the screen
preview to 1 second. Added READ_CALL_LOG permission to the manifest.

Commit:
Sticky Notification off by default, and spash screen off by default.

Fixed Disconnect and Exit to work on all Android devices, System.exit(0)
is unreliable.

Added edge glow and fade out/in animation to watch previews.

Commit:
Added starting service please wait toast, improved shutdown preference
functionality to accommodate the recent apps menu, and changed the main
service toggle to an onCheckedChangeListener.

Commit:
New light theming.

Improved watch preview animations.

Improved widget list layout, font size, and alignment.

Improved Service management specifically with regard to initial installs
prior to watch selection.

Commit:
Layout file from previous commit.

benjymous commented:
Cool - I'll have a look when I get a free moment :)

Is it just UI stuff you've changed?

nseidm1 commented:
At first it was mostly UI stuff, now almost everything has changed :-( The biggest changes are in the service, Idle and Protocol have been turned into singletons, and static references have been removed. The CE group may want to take a peek at onStartCommand in the Service and see how I'm using it, fun stuff to avoid statics. 

Everything has been fully optimized for proper memory management, and efficient garbage collection.

Commit:
Log class to enable and disable log calls. Substantially modified Theme
interface, swapping the TabHost for ActionBar Tabs. Improved handling of
the downloaded theme list, including visual feedback with an
indeterminate progress bar. Improved animations in the main activity
status.

Commit:
Fixed non backwards compatible import.

Commit:
Fixed up affordance in new themes activity. Replaced deprecated
notification constructor with NotificationCompat in the support library.

Commit:
Added holo blue gradient to main widget list divider, and red gradient
divider to child views. Removed loading animations, leaving only the
preview change fade out/in animation. Added failed to load theme list
handling via a dialog.

Commit:
Added Noah Edition to about in settings. Added left padding to device
selection list items.

Commit:
Applied consistent code formatting project wide. Increased resolution of
watch preview images.

Commit:
Added please wait string to the Statistics dialog prior to the
availability of data. Added hardwareAccelerated flag to the application
tag in the manifest. Moved main animation management to the activity
rather than the views in the fragments. Thread the starting of the
service, and only update the Statistics text if the dialog is showing;
this should reduce UI lag. Main animations will only play if the Service
is running, the service does a lot of stuff on the UI thread, and
without investigating the service, there's no easy way accommodate
things.

Commit:
Remove lock on Statistics dialog, it'll just update like it did before.
Small delay in Animation start in main activity.

Commit:
Added slide in from right animation to widget list child items. Added
shake animation when toggling the service. Improved handling service
button and action bar status text.

Commit:
Service button shake tweak.

Commit:
Removed progress dialog during theme download in favor of action bar
progress bar; removes change for illegal argument exception. Tweaks to
main animations, and service state toggle button.

Commit:
Switched to ActionBarSherlock 4.2.0. Animation tweaks.

Commit:
Switched notification icon back to original. Fixed potential null
pointer associated with getting recent best location. Fixed race
conditions with the clients array in the service, by switching to a
vector array, and synchronizing notifyClients method on mClients.

Commit:
Changed expandable list children to fade in animation.

Commit:
Changed Protocol to a singleton for improved memory management. Removed
wake locks.

Commit:
Removed static references to fields in the Service in favor of the
intent based event bus design pattern.

Commit:
Turned Idle into a singleton, facilitation destruction in onDestroy of
the main service. Removed a lot of static method calls.

Commit:
Removed static modified from field in Service.

Commit:
Removed two prferences, "Notify Watch on Connect", and "Display Splash
Screen". I personally, and users, have reported issues.

Commit:
Manifest bump.

Commit:
Synchronize enqueue, and null check in onStartCommand.

Commit:
Possible deadlocks?

Commit:
Synchronized send notifyClients intent request.

Commit:
Incoming call contact photo causes watch to reboot on 1.3 firmware.
Removed contact photo from canvas painting. The resolution really
doesn't warrant the contact's photo and nothing more than name and
number are more than fine.

Commit:
Update data and tme formats automatically on watch connect.

Commit:
No need to synchronize those methods, onStartCommand is inherently
synchronized.

Commit:
Re-added the incoming call contact photo, and added a single thread pool
executor to the service for input/output stream interaction.

Commit:
Fixed duplicate call to endRinging. Remove call to update screen during
call vibration, it overloads the watch causing it to crash; no need to
update the watch screen it's not changing at all. Changed names of
transmit and receiver threads in the service just to be clear about
their function.

Commit:
Re-added strategic wakelock allowing the device to sleep.

Commit:
Prevent service from starting due to receivers and observers. Would be
better to tie the monitors to the lifecycle of the service, this should
suffice?

Commit:
Dam, without the wakeLock things simply get out of sync when the cpu
shuts down.

Commit:
Notification, MediaControl, and Monitors are now singletons tied to the
lifecycle of the service. The service now reconnects almost instantly,
with a fully optimized connection process! WakeLock is now handled more
gracefully hopefully allowing a deep sleep to occur. Lingering static
values have been removed allowing for proper garbage collection, and
additionally facilitating an almost instant reconnection process. Lots
of cleanup....

Commit:
Tie the three managers, AppManager, WidgetManager, and
ApplicationManager to the lifecycle of the service for improved memory
management and performance.

Commit:
Simplified the cleanup process, and further encourage loading after the
application has instantiated. Provides a faster manager application
launch time, and a faster service disconnect time.

Commit:
The repeating runnable loves to cause the watch crash.

Commit:
Improved AccessibilityCheck Service interaction for statistics dialog.

Commit:
Removed 4.2 method call.

Commit:
Tweaked wakelock management, and remove unnecessary async block.

Commit:
Increased default packet wait to 25 milliseconds.

Commit:
Disabled media app menu button, investigation is needed by back doesn't
work properly when in the menu.

Commit:
Fixed idle pages getting out of sync when switching using the manager
app, then switching using the top right watch button.

Commit:
Fixed the media app menu, now quite reliable.

Commit:
Revert "Fixed the media app menu, now quite reliable."

This reverts commit d58f98d09337a6723cdc4175ed7b8d9acd351119.

Commit:
The MediaPlayer app menu will remain disabled, the CE team will take a
peek if desired. Disabled AppLaunchMode settings option, anything other
than Popup causes issues. Set the default wait of the incoming packet
thread to the same as the outgoing in Protocol.

Commit:
Minor text fixes.

Commit:
Updated BugSense jar.

Commit:
Removed manual launch of AccessibilityService to support Gingerbread and
lower devices.

Commit:
Simplified service management toggle button. Cleanup of auto start
functionality to prevent spontaneous service starting. Lazy handling of
draw issue in ActionApp, pass exception up the chain.

Commit:
Binding a service to an activity makes for headaches and is typically
not requires except for IPC. Also the state of a service is not on and
off, so an on/off button is not best suited. Two buttons are more
analagous to the lifecycle of a service, in turn requesting that the
service either startup or shutdown.

Commit:
Improved management of the service via the start and shutdown buttons.

Commit:
Further improved management of the service and the start and shutdown
buttons.

Commit:
Cleanup, and test battery weather poll technique. Everything but the new
poll technique will definitely be merged to master.

Commit:
New global theme.

Commit:
Title mod for preferences and tests, and main animation tweak.

Commit:
Added action bar menu to enable and disable animations.

Commit:
Added onClick response to the watch previews.

Commit:
Added preference to show the sms message only, without the intro
notification.

Commit:
Fixed bug with silent mode toggling. Changes some timing values.

Commit:
Cleanup DeviceSelection including device auto connect after device
selection. Changed watch previews glow to red. Other minor cleanup.

Commit:
Added contact's name to sms notifications message display.

Commit:
Updated Italian translations. Improved async refreshing of weather data.

Commit:
Fixed toggle app button becoming non functional after returning from the
MediaApp's menu. Fixed reliable/consistent switching of apps. Fixed
toggle wifi menu option to update the idle page reflecting the new state
of the wifi hardware. Sticky notifications now cannot be enabled. Also
disabled the packet timeout preference, fixed now at 30 milliseconds.

Commit:
Fixed race condition associated with closing an activity before a thread
is finished.

Commit:
Fixed enable/disable wifi toggle for the Actions in both application and
idle mode.

Commit:
StickNotifications and packet delay are now fixed values.

Commit:
Increased haptic feedback. Fixed toggle wifi feature for both app mode
and idle mode. Removed unnecessary async, wasn't needed.

Commit:
Removed entire Protocol queue in favor of consolidating the queue
directly into the Service. This should facilitate much better recovery
from IOException, in turn allowing the reconnection process to occur and
continue where it left off. Otherwise everything will gracefully halt,
clean, and shutdown in the case the Service is requested to stop.

Commit:
Service reconnect cleanup less objects, focus on more cleanup in
onDestroy instead. Readded message queue statistics to manager
statistics dialog.

Commit:
Ensure that the Notifications queue is cleared if the service's
connection is lost, notifications likely will not be pertinent when
connection is restored in the future.

Commit:
New seamless reconnection pattern. Message queue, and notification queue
will both block until either a connection is re-established or the
Service is destroyed. This pattern also allows for notifications to
queue even if the service isn't connected avoiding missed notifications,
which some users have reported.

Commit:
Removed the repeating sms test. Cleanup of Service field declarations.
Added some comments.

Commit:
Ensure that an IDLE_NEXT_PAGE button press has the system in IDLE mode.
Otherwise updateIdlePages will not refresh properly. Typically this
isn't an issue, but a race condition sometimes occurs.

Commit:
Revert "Ensure that an IDLE_NEXT_PAGE button press has the system in IDLE mode. Otherwise updateIdlePages will not refresh properly. Typically this isn't an issue, but a race condition sometimes occurs."

This reverts commit 0dee12c8c3c7782abb5b5f7b45b33688bb04f2e3.

Commit:
Fixed next page button when in app mode, yay :)

Commit:
Manifest bump, previous commit fixing next page button in app mode
warrants an update.

Commit:
Null checks, service onDestroy stability.

Commit:
Removed emulated bluetooth media controls, they don't work. Tested the
regular method with Play Music, Spotify, and Pandora; works fine.

Commit:
Manifest bump.

Commit:
Revert "Removed emulated bluetooth media controls, they don't work. Tested the regular method with Play Music, Spotify, and Pandora; works fine."

This reverts commit a2df8dfdb94e70bdb68ca8d1c7d7ea16d2bced7a.

Commit:
Ugg, reverted the bluetooth control. I dunno. Added an Idle update call
when toggling inverse media controls.

Commit:
Service shutdown when bluetooth hardware is turned off. Restructured the
notfication queue to be consistent with the design pattern of the
message queue. Tweaks to how the queues shutdown when the service is
shutdown from a blocked state.

Commit:
Excellent cleanup of the service. Lifecycle methods at the top, static
classes at the bottom. Added some comments, and improved field
declaration organization. I'm now very satisfied with the design of the
three main async components, the notifications queue, message queue, and
read from watch thread.

Commit:
Stability assurance catching. Replaced ScheduleExecutorService with a
standard ThreadPoolExecutor.

Commit:
Fixed forced weather update to not only update Idle pages, but also App
pages if applicable.

Commit:
Manifest bump.

Commit:
Voice search will now show on the watch as not installed accordingly in
the quick actions.

Commit:
Possible fix for reported clock on every page issue? Possible stability
improvement always updating the buffer size before painting a screen,
has it been a bufferOverrun scenario?

Commit:
Likely fixed clock on every page issue. Also likely fixed button mapping
problems. General cleanup and method consolidation.

Commit:
Partial revert.

Commit:
Removed call to updateLcdDisplay without actually sending an image,
possible null issue on the watch?

Commit:
Resolved non functioning buttons in application mode?

Commit:
Removed application tests, may causes issues and not useful for the gen
pub. Most important change is in ApplicationBase, which resolves app
navigation issues when the QuickActions app is enabled.

Commit:
Fixed next page button when in app (ie. QuickAction) non Idle mode.

Commit:
Improved reconnection process, more is destroyed.

Commit:
More cleanup during the reconnection process.

Commit:
Ehh.

Commit:
Less cleanup when reconnecting, why destroy everything only to re-create
it?

Commit:
Manifest bump.

Commit:
Modified reconnect flow.

Commit:
A little backtrack, a little sidestep, a little forward.

Commit:
Synchronization, Idle updating, Service lifecycle.

Commit:
New graphics, inverted background.

Commit:
Minor issue enabling/disabling apps and refreshing.

Commit:
Fixed app icon in About page, and changed watch preview halo to holo
blue.

Commit:
Stack based mode management testing.

Commit:
Watch state management cleanup, organization, simplification, and
improvement. Improved disconnect and exit.

Commit:
Minor cleanup and missing break bug in switch statement.

Commit:
Improved, consolidated, and simplified mode management, specifically end
mode functionality. See: NavigationManagement class. Work on call screen
button features.

Commit:
Minor tweaks to watch mode management. Active bluetooth enabled state
management, request to turn on bluetooth if disabled.

Commit:
Import cleanup.

Commit:
Fixed shutdownRequested flag.

Commit:
Auto bluetooth enable when starting the service, fix auto start on boot,
other minor cleanup.

Commit:
Fixed next page when in app mode handling the watch state properly.

Commit:
Removed image viewer, who would want to use it anyway. Consolidated
AlarmReceiver calls into the recursive runnable in the service, there
was another call in the Monitors that used an rtc wakeup alarm. Manifest
cleanup. Sync mod in Idle. Notification cleanup, removed lingering
sticky notifications code.

Commit:
Re-added image viewer class, testing issues, mods in the NE make it work
just fine.

Commit:
On reconnection stop app mode.

Commit:
Add single vibration to image viewer feature.

Commit:
Navigation and idle refresh tweaks, improvement?

Commit:
Improved navigation on reconnect.

nseidm1 committed:
Fixed default notification timeout for API notifications.

nseidm1 committed:
Turning off bluetooth will no longer stop the service. I removed the
wakelocks in the service, so be in a perpetual state of connecting will
not affect battery life. The service will automatically enable bluetooth
sooooo, if you actually want to turn off your bluetooth hardware you'll
want to turn off the service!

nseidm1 committed:
Fixed crash related to calls. It was attempting to retrieve a name for a
telephone number not in your phonebook.

Cleanup regarding Yahoo weather situation. You're status will now inform
you Yahoo is deprecated, and  Wundeground needs to be used.

Bluetooth will now only be automatically enabled when you press the
start service button.

heisingrf committed:
Update Call.java

Updated the name/number text to use the preference stored under notifications.

nseidm1 committed:
Merge pull request #3 from heisingrf/patch-2

Update Call.java

Commit:
Fixed merge pull request issue.

Commit:
Resolved incoming call crash condition.

davemclaughlin committed:
Fixed clock not being set on the watch

Clock on the watch now works but bitmap still wrong on phone. Still to
be sorted.

nseidm1 committed:
Merge pull request #4 from davemclaughlin/master

Fixed clock not being set on the watch

nseidm1 committed:
Minor changes.
